qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -133,6 +141,46 @@ public CompletableFuture<List<CodeAction>> doCodeActions(Template template, Code
 		return CompletableFuture.completedFuture(codeActions);
 	}
 
+	private void doCodeActionToCreateProperty(Template template, Diagnostic diagnostic, QuteErrorCode errorCode,
+			List<CodeAction> codeActions, QuteTemplateGenerateMissingJavaMember resolver, SharedSettings settings) {
+		JsonObject data = (JsonObject) diagnostic.getData();
+		String missingProperty = data.get("property").getAsString();

```

Comment:
If client doesnt support data, the data will be null. You must take care of this npe and get in this case the resolved type. For the last part you could do that in an another pr since vscode and eclipse ide support data